samcli/commands/_utils/template.py

```diff
@@ -161,6 +156,31 @@ def _update_relative_paths(template_dict, original_root, new_root):
     return template_dict
 
 
+def _transform_dict(template_dict, path, transformation, context=None):
+    def visit(sub_dict, sub_path):
+        if not sub_path:
+            return sub_dict
+
+        next_prop = sub_path[0]
```

Comment:
Isn't this the current property rather than the next? `property` or `prop_name` might be better names